src/GafferOSL/OSLShader.cpp

```diff
@@ -1258,11 +1397,30 @@ static IECore::ConstCompoundDataPtr metadataGetter( const std::string &key, size
 				// If you specify conflicting metadata on the different parameters you may get inconsistent results.
 				CompoundData *prevData = parameterMetadata->member<CompoundData>( nameWithoutSuffix );
 				CompoundDataPtr data = convertMetadata( parameter->metadata );
+
+				CompoundDataPtr positionData = convertMetadata( positionsParameter->metadata );
+				CompoundDataPtr valuesData = convertMetadata( valuesParameter->metadata );
+				CompoundDataPtr basisData = convertMetadata( basisParameter->metadata );
 				if( prevData )
 				{
 					data->writable().insert( prevData->readable().begin(), prevData->readable().end() );
 				}
 
+				if(
+					( positionData->member<StringData>( "widget" ) && positionData->member<StringData>( "widget" )->readable() != "null" ) ||
+					( valuesData->member<StringData>( "widget" ) && valuesData->member<StringData>( "widget" )->readable() != "null" ) ||
+					( basisData->member<StringData>( "widget" ) && basisData->member<StringData>( "widget" )->readable() != "null" )
+				)
```

Comment:
I don't know what's wrong with this logic, but I'm still not getting any spline widgets on any of the DL shaders. What works for me is far simpler :

```
CompoundDataPtr data = convertMetadata( parameter->metadata );
+ data->writable().erase( "widget" );
```

We don't have any support for `widget` values that would be valid for ramp, so this seems fairly reasonable, unless we think someone would really want to hide one?

Reply:
I'm fine removing the hidability of spline widgets, they do seem like an odd thing to want to include outside the users view. Fixedin https://github.com/GafferHQ/gaffer/pull/5432/commits/b768452d86ce5cc472207dd1bb6ac954f7c73918